application/src/main/java/com/sanctionco/thunder/util/HashUtilities.java

```diff
@@ -0,0 +1,38 @@
+package com.sanctionco.thunder.util;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+import org.apache.commons.codec.binary.Hex;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Provides utility methods related to hashing String values.
+ */
+public class HashUtilities {
+  private static final Logger LOG = LoggerFactory.getLogger(HashUtilities.class);
+
+  /**
+   * Performs a hash of the given plaintext using the given hash algorithm.
+   *
+   * @param hashAlgorithm The hash algorithm to use when hashing.
+   * @param plaintext The plaintext to hash.
+   * @return The hashed value of the plaintext.
+   */
+  public static String performHash(String hashAlgorithm, String plaintext) {
```

Comment:
Can we make the hashing algorithm an enum? This might make it easier to enforce a particular algorithm.

Reply:
Yes good idea. I'll do that in the next PR